runtime/jcl/common/shared.c

```diff
@@ -585,8 +585,11 @@ getCpeTypeForProtocol(JNIEnv *env, const char *protocol, jsize protocolLen, cons
 	pathChars = *correctedPathPtr;
 	pathLen = (jsize)strlen(pathChars);
 	if (0 == strncmp(protocol, "file", 5)) {
-		if ((NULL != strstr(pathChars, "!/")) || (NULL != strstr(pathChars, "!\\"))) {
-			/* file is a fat jar */
+		char *endsWith = (char *)(pathChars + (pathLen-4));
+		if ((0 == strncmp(endsWith, ".jar", 4)) || (0 == strncmp(endsWith, ".zip", 4))
+			|| (NULL != strstr(pathChars,"!/")) || (NULL != strstr(pathChars,"!\\"))
```

Comment:
What ensures that `pathLen >= 4` so `pathLen - 4` is a reasonable offset?

The type of `endsWith` should be `const char *` so no cast is necessary.

Missing spaces after `,` and both sides of `-`.